parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -201,6 +210,57 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return internalReader.nextKeyValue();
   }
 
+  /**
+   * Reads the next batch of rows. The caller needs to check the returned batch size with {@link parquet.vector.RowBatch#size()}.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+  public RowBatch nextBatch(RowBatch previous) throws IOException {
+    MessageType requestedSchema = readContext.getRequestedSchema();
+    List<ColumnDescriptor> columns = requestedSchema.getColumns();
+    int nColumns = columns.size();
+    ColumnVector[] columnVectors;
+    if (previous == null || previous.getColumns() == null) {
+      previous = new RowBatch();
+      columnVectors = new ColumnVector[nColumns];
+      for (int i = 0; i < nColumns; i++) {
+        ColumnVector columnVector = ColumnVector.createVector(columns.get(i));
+        MessageType columnSchema = new MessageType(requestedSchema.getFieldName(i),
+            requestedSchema.getType(i));
+        readVector(columnVector, columnSchema);
+        columnVectors[i] = columnVector;
+      }
+    } else {
+      columnVectors = previous.getColumns();
+      for (int i = 0; i < nColumns; i++) {
+        ColumnVector columnVector = columnVectors[i];
+        MessageType columnSchema = new MessageType(requestedSchema.getFieldName(i), requestedSchema.getType(i));
+        readVector(columnVector, columnSchema);
+      }
+    }
+
+    previous.setColumns(columnVectors);
+    return previous;
+  }
+
+  /**
+   * Read next batch of ColumnVector
+   * ColumnVector holds the bytes of the vector
+   * ColumnVector.decode() materializes the vector values
+   * @param vector the vector to fill
+   * @param column the requested column
+   * @throws java.io.IOException
+   */
+  private void readVector(ColumnVector vector, MessageType column)
+  {
+    try {
+      internalReader.nextBatch(vector, column);
+    } catch (Exception e) {
+      LOG.info("Get exception when reading batch: " + e.getMessage());
```

Comment:
don't swallow the exception here, you should propagate that to the caller (and the application eventually)